packages/utils/src/cache/storages/indexeddb.ts

```diff
@@ -0,0 +1,177 @@
+
+interface IndexedDBDatabase extends DBSchema {
+  [key: StoreName]: {
+    key: string;
+    value: CacheItem;
+  }
+}
+
+/**
+ * Literal type for dynamic store name casting to idb
+ */
+type StoreName = `[${StoragePrefix.DEFAULT}][${string}]`;
+
+/**
+ * Returns Error message when unable to connect indexedDB
+ * Database and Store are same name. Version is 1. So don't try to log a real database config.
+ * @param message error message
+ * @param dbName database name
+ * @returns Error message
+ */
+const errorMessage = (message: string, dbName: string): Error => {
+  return new Error(`Unable to connect to indexedDB.\nAttempt connect database is name: ${dbName}. version: 1. store: ${dbName}\n ${message}`);
+};
+
+/**
+ * Stores data in indexedDB for use across multiple sessions.
+ */
+export class IndexedDBStorage implements CacheStorage {
+  /**
+   * Prefix for database name, store name, and key of all items
+   * to avoid database to clash with other storages.
+   */
+  protected dbName: StoreName;
+
+  /**
+   * A connection to a indexedDB database, use for open transaction or idb api
+   */
+  private db: IDBPDatabase<IndexedDBDatabase> | undefined;
+
+  /**
+   * Database Version.
+   */
+  private version = 1;
+
+  /**
+   * Internal cache object
+   */
+  protected cache: CacheMap | null | undefined;
+
+  /**
+   * Cache ready to use
+   */
+  protected ready: Promise<boolean> | null = null;
+
+  /**
+   * Constructor
+   * @param name for database name and store name
+   */
+  constructor (name: string) {
+    this.dbName = `[${StoragePrefix.DEFAULT}][${name}]`;
+    void this.open();
+  }
+
+  /**
+   * Set a item against a key to this storage
+   * @param key item key
+   * @param value item value
+   * @returns {void}
+   */
+  public async set (key: string, value: CacheItem): Promise<void> {
+    await this.ready;
+    const itemKey = getItemKey(this.dbName, key);
+    const item = { ...value, itemKey };
+    this.cache?.set(itemKey, item);
+    await this.db?.put(this.dbName, item, itemKey);
+  }
+
+  /**
+   * Returns a item in this storage that matched by the key.
+   * @param key item key
+   * @returns cache item or null
+   */
+  public async get (key: string): Promise<CacheItem | null> {
+    await this.ready;
+    const itemKey = getItemKey(this.dbName, key);
+    return this.cache?.get(itemKey) || null;
+  }
+
+  /**
+   * Remove a item against a key to this storage
+   * @param key item key
+   * @returns {void}
+   */
+  public async remove (key: string): Promise<void> {
+    await this.ready;
+    const itemKey = getItemKey(this.dbName, key);
+    this.cache?.delete(itemKey);
+    await this.db?.delete(this.dbName, itemKey);
+  }
+
+  /**
+   * Clear all items in this storage
+   * @returns {void}
+   */
+  public async clear (): Promise<void> {
+    await this.ready;
+    this.cache?.clear();
+    await this.db?.clear(this.dbName);
+  }
+
+  /**
+   * Restore all values into memory cache
+   * @returns Promise void
```

Comment:
```suggestion
   * @returns {void}
```

Reply:
Updated to following your suggestion. And do this to other comment.